src/Collections/MemoryConfigCollection.php

```diff
@@ -215,6 +215,10 @@ public function getAll()
      */
     public function update($class, $name, $value)
     {
+        user_error(
+            'Config::inst()->update() is deprecated. Please use YAML configuration, Config::modify()->merge() or ->set()',
+            E_USER_WARNING
```

Comment:
Yes. Damian's suggestion was to throw a fatal error here and not continue with `merge` - will wait to see if that's still what he wants before I update this. Thanks :)